src/evaluate/evaluator/base.py

```diff
@@ -236,7 +239,13 @@ def prepare_data(self, data: Union[str, Dataset], input_column: str, label_colum
             raise ValueError(
                 "Please specify a valid `data` object - either a `str` with a name or a `Dataset` object."
             )
-        data = load_dataset(data) if isinstance(data, str) else data
+        if isinstance(data, str):
+            if not kwargs.get("data_split"):
+                split = choose_split(data)
+                logger.warning(f"Dataset split not defined! Automatically evaluating with split: {split.upper()}")
+            else:
+                split = kwargs['data_split']
+            data = load_dataset(data)[split]
```

Comment:
If you use `split=` in `load_dataset` you can avoid downloading files we don't need. With that we could also make use of the [slice-split](https://huggingface.co/docs/datasets/loading#slice-splits) functionality in `datasets` and for example pass `data_split="validation[:100]"`:
```suggestion
            if data_split is None:
                data_split = choose_split(data)
                logger.warning(f"Dataset split not defined! Automatically evaluating with split: {split.upper()}")
            data = load_dataset(data, split=data_split)
```